guides/authentication.md

```diff
@@ -212,7 +219,113 @@ client.loginWithOAuth({
 });
 ```
 
-## More
+### More
 
 Here is an article describing more in detail about the authentication process:
 https://doc.cognitedata.com/dev/guides/iam/external-application.html#tokens
+
+## Azure AD auth flow
+
+Azure AD auth flow can be used via the same `loginWithOAuth` method of `CogniteClient`,
+but different `options` structure has to be passed. You'll find what options structure has to be provided in examples below.
+Besides, here list of pre-requirements, that need to be in place to use Azure AD authentication flow:
+
+ - CDF project, which you're trying to log in, has to be configured to accept tokens issued by Azure AD (to be related to the cluster, that accepts Azure AD tokens, `bluefield` for instance)
```

Comment:
I believe default cluster also accepts them. Only difference is that you have to ask auth team to enable it for the project

Reply:
That's correct, the interesting bit would be that some clusters _only_ support this flow (all new ones), otherwise it depends on the projects you have access to in the cluster

Reply:
This section has been rewritten